### DISCUSSION TITLE
Use modifier keys as their own key when pressed alone

### DISCUSSION THREAD

**Discussion by SeeThruHead** [cajhin/capsicain#109] [General]
Been looking through the manual and I can’t see what I’m looking for. 

The one thing I can’t do on windows that I need desperately is the ability to reuse modifier keys as stand alone keys when not pressed with a keg to modify. 

For example. 

Left shift -> (
Right shift -> )

ctrl -> esc

when I press and release control without pressing any other key I want it to send escape. 

When I press and release left shift without pressing any other key while shift is held down it should send left paren

You get the idea. 

This is something that I use Karabiner elements for on Mac that I can’t live without.

**Comment by cajhin** [comment]
check REWIRE.
What you describe as "stand alone" is what I call tapping.
REWIRE LSHF LSHF ESC
should do what you want (left shift is left shift, but ESC when tapped without any other key involved)

Note that I don't like that personally. Adds complexity to something that should be very basic (the shift key). If you press shift, then change your mind and release it, or if you type very fast and release shift a milisecond too early, you get ESC and who knows what happens then...)

update: forgot to mention, sorry you can't rewire to ( because that's not a key (it's the combo shift-9 or whatever your active keyboard layout is). Not sure if it is possible to define a combo "tapped shift to shift-9"...

update2: closest I could figure out is double tap Shift for '(' :
COMBO LSHF [.T] > moddedKey( 8 + .&)
but you'll have a problem if after the "double tap for (" you want to type Shift-something, because the 3rd press of shift will trigger a second (
See if you can live with that...

**Reply by SeeThruHead** [reply]
I’ve got ten years of muscle memory coding this way. Can’t change now. Also I like it a lot and don’t want to change really. 

Lshift = ( and rshift = )

is the most important. Too bad.

**Comment by cajhin** [comment]
I get it, I would never type this comment on a Macbook if I didn't have Karabiner.
Out of curiosity, I was looking for a way to configure your ().
Came up with a slightly hacky workaround that I have only cursory tested for unwanted side effects.

If you're interested, I can check in the experimental exe and you try it out. Interested?

**Comment by SeeThruHead** [comment]
Would be happy to try it.

**Comment by cajhin** [comment]
ok, copy the exe v98test from here: 
https://github.com/cajhin/capsicain/tree/master/x64/Release

put this in your ini:

```
REWIRE lshf lshf WEBFORWARD
REWIRE rshf rshf WEBBACK
COMBO WEBFORWARD [] > moddedkey(9+&)
COMBO WEBBACK [] > moddedkey(0+&)
```

See if it works for you, and either way, please give me feedback.

This isn't quite bullet proof, a bit hacky and I cannot play all possible scenarios in my head.

The short version is: so far, when you REWIRE a tapped key to another key, then that key was not processed anymore, so no combos were possible.
With v98test, for the special case of tapped/rewired, the result IS tested against all combos.
So, I rewire tapped shift to useless keycodes ("WEBFORWARD/-BACK"), 
then I have combos for these keycodes (WEBFORWARD -> Shift-8)

...not proud of that, just a quick hack to maybe get something accurate going...
